Sources/MongoSwift/MongoCursor.swift

```diff
@@ -152,27 +152,31 @@ public class MongoCursor<T: Codable>: CursorProtocol {
      *
      * If this cursor is not tailable, this method will exhaust it.
      *
-     * If this cursor is tailable, `all` will only fetch the currently available results, and it
+     * If this cursor is tailable, `toArray` will only fetch the currently available results, and it
      * may return more data if it is called again while the cursor is still alive.
      *
      * - Returns:
-     *    An `EventLoopFuture<[T]>` evaluating to the results currently available to this cursor or an error.
+     *    An `EventLoopFuture<[T]>` evaluating to the results currently available in this cursor, or an error.
      *
      *    If the future evaluates to an error, that error is likely one of the following:
      *      - `CommandError` if an error occurs while fetching more results from the server.
      *      - `LogicError` if this function is called after the cursor has died.
      *      - `LogicError` if this function is called and the session associated with this cursor is inactive.
      *      - `DecodingError` if an error occurs decoding the server's responses.
      */
-    internal func all() -> EventLoopFuture<[T]> {
+    public func toArray() -> EventLoopFuture<[T]> {
+        guard self.isAlive else {
```

Comment:
without this, we would never error even if you call this on a closed cursor and would always return an empty array, because `self.cached.clear()` would just return nil if the cursor is closed.

Reply:
Good catch. One thing to note is that calls to `isAlive` and `cached.clear` should probably both only occur while the cursor lock is held, which is making me think we may have to move the cache logic into the internal cursor after all.

Reply:
I've moved the caching logic into the common cursor type, and added a static property on `MongocCursorWrapper` as we discussed to indicate whether we need to cache the first document or not